plugins/catalog-import/src/components/StepPrepareCreatePullRequest/StepPrepareCreatePullRequest.tsx

```diff
@@ -82,10 +84,11 @@ export function generateEntities(
     kind: e.kind!,
     metadata: {
       ...e.metadata,
-      name: componentName,
+      name: entityName,
     },
     spec: {
       ...e.spec,
+      type,
```

Comment:
Beware: The `/analyze-entities` endpoint does create a single `Component` at date. But in the future, I assume it could also create entities such as `API` where you don't want to replace the `type`. Even my choice to override the name of all entities is questionable in that light.

An alternative here could be here to have a form for each entity that could be configured. But that would require a larger restructuring of the form. That makes me think about a form where one could also add new entities and that is just pre-populated with the entities that were discovered by the backend. Such forms could e.g. be generated based on the JSON-schemas of the entities.

Reply:
Yes, I thought about that. i actually have a good use-case : a monorepo which contains multiple front-ends directories (websites) and multiple microservices directories (services). The endpoint should ideally detect:
* all services
* all websites
* APIs which are provided and consumed by the services (much more complex)

But as long as the endpoint does no such thing, I think it's an improvement to allow the user to pick the type of the "main component" of this repository. Does not prevent him from then adding extra catalog-info.yaml files in the repository to cover complex monorepos.

In summary : I suggest we keep this change for now. What do you think?